### PR TITLE
Update module structure

### PR DESCRIPTION
**Pull request by Chuhaa**
# Description

Update module structure

Fixes # ([issue](https://github.com/wso2-enterprise/choreo/issues/15334))

One line release note: 
- Update module structure

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Test Configuration**:
* Ballerina Version: 2201.2.0
* Operating System: Ubuntu
* Java SDK: 11

# Checklist:

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: salesforce/client.bal [hunk 1]
```diff
@@ -69,6 +65,250 @@ public isolated client class Client {
...
+
+    // External function for the conversion of stream
+    isolated function streamConverter(stream<record {}, error?> data, typedesc<record {}> returnType) returns
+    stream<record {}, error?>|error = @java:Method {
+        'class: "io.ballerinax.salesforce.ReadOperationExecutor"
+    } external;
+
     ///////////////////////////////////////////// DEPRECATED ///////////////////////////////////////////////////////////
 
     //Describe SObjects
```
Comment (sachinira) [inline salesforce/client.bal:314]: Check and update the deprecated comment in all deprecated operations.

Comment (Chuhaa) [reply salesforce/client.bal:314]: Updated in https://github.com/ballerina-platform/module-ballerinax-sfdc/commit/0997adbe6d633d3c7bc158f492d44430435bd7fb

File: salesforce/client.bal [hunk 2]
```diff
@@ -69,6 +65,250 @@ public isolated client class Client {
...
+    isolated remote function getResources(@display {label: "API Version"} string apiVersion)
+                                                    returns @display {label: "Resources"} map<string>|error {
+        string path = utils:prepareUrl([BASE_PATH, apiVersion]);
+        json res = check self.salesforceClient->get(path);
+        return toMapOfStrings(res);
+    }
+
+    # Lists the Limits information for your organization.
+    #
+    # + return - `OrganizationLimits` as map of `Limit` if successful. Else, the occurred `Error`.
```
Comment (sachinira) [inline salesforce/client.bal:138]: This should be error

File: salesforce/client.bal [hunk 3]
```diff
@@ -69,6 +65,250 @@ public isolated client class Client {
...
+    isolated remote function getPlatformAction() returns @display {label: "sObject Basic Information"}
+                                                SObjectBasicInfo|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, PLATFORM_ACTION]);
+        return check self.salesforceClient->get(path);
+    }
+
+    //Describe Organization
+    # Lists summary details about each REST API version available.
+    #
+    # + return - List of `Version` if successful. Else, the occured Error.
```
Comment (sachinira) [inline salesforce/client.bal:117]: This should be error

File: salesforce/client.bal [hunk 4]
```diff
@@ -69,6 +65,250 @@ public isolated client class Client {
         }
     }
 
+    //Describe SObjects
+    # Get metadata for your organization.
+    #
+    # + return - `OrganizationMetadata` record if successful or else `sfdc:Error`
```
Comment (sachinira) [inline salesforce/client.bal:71]: This should be error. Check all places

Comment (Chuhaa) [reply salesforce/client.bal]: Fixed in https://github.com/ballerina-platform/module-ballerinax-sfdc/commit/666dfedf90963e702c3e32712fa8e2e41f087f2e

File: salesforce/client.bal (hunk 2 above)
Comment (Chuhaa) [reply salesforce/client.bal]: Fixed in https://github.com/ballerina-platform/module-ballerinax-sfdc/commit/666dfedf90963e702c3e32712fa8e2e41f087f2e

File: salesforce/client.bal (hunk 3 above)
Comment (Chuhaa) [reply salesforce/client.bal]: Fixed inhttps://github.com/ballerina-platform/module-ballerinax-sfdc/commit/666dfedf90963e702c3e32712fa8e2e41f087f2ehttps://github.com/ballerina-platform/module-ballerinax-sfdc/commit/666dfedf90963e702c3e32712fa8e2e41f087f2e

**Review by sachinira — APPROVED** [review]

**Pull request merged**